distributed/deploy/ssh.py

```diff
@@ -112,12 +117,17 @@ async def start(self):
                 set_env,
                 self.remote_python or sys.executable,
                 "-m",
-                self.worker_module,
+                "distributed.cli.dask_spec",
                 self.scheduler,
-                "--name",
-                str(self.name),
+                "--spec",
+                "'%s'"
+                % json.dumps(
+                    {
+                        "cls": self.worker_class,
+                        "opts": {**self.kwargs, "name": self.name,},
```

Comment:
Is the `"name"` parameter here necessary? This is causing errors, probably because `CUDAWorker` will create multiple workers at once, one per GPU:

```python
distributed.nanny - ERROR - Failed while trying to start worker process: Unexpected response from register: {'status': 'error', 'message': 'name taken, 0', 'time': 1593107200.1947834}
distributed.nanny - ERROR - Failed to start worker
Traceback (most recent call last):
  File "/datasets/pentschev/miniconda3/envs/rn-102-0.15.0b200625/lib/python3.7/site-packages/distributed/nanny.py", line 766, in run
    await worker
  File "/datasets/pentschev/miniconda3/envs/rn-102-0.15.0b200625/lib/python3.7/site-packages/distributed/core.py", line 290, in _

    await self.start()

  File "/datasets/pentschev/miniconda3/envs/rn-102-0.15.0b200625/lib/python3.7/site-packages/distributed/worker.py", line 1093, in start
    await self._register_with_scheduler()
  File "/datasets/pentschev/miniconda3/envs/rn-102-0.15.0b200625/lib/python3.7/site-packages/distributed/worker.py", line 847, in _register_with_scheduler
    raise ValueError("Unexpected response from register: %r" % (response,))
```

The `--name` parameter was also being used before but wasn't causing any issues, I don't know why that happens with the spec. I tried just removing `"name": self.name,` and that alone suffices to resolve the issue with `CUDAWorker`.

Reply:
Typically the cli commands will take a name parameter and expand it out to name-0, name-1, and so on.  I suspect that this is what the dask-cuda cli was doing before?  Maybe we could replicate that logic in dask-cuda?

Reply:
Yes, we're doing that. I now found that the name we're passing is actually an integer instead of a string, that's why it fails now, it was safe before, even though `name` is indeed expected to be a string. I will fix that on our end, but maybe it's still worth to keep the `str(self.name)` in case someone does the same mistake we did?

Reply:
Actually I can't fix it on our end. We never pass the `name` argument from `SSHCluster` to the `Worker` class. We would either need to keep the `str(self.name)` or add a way to pass it via `SSHCluster`.

Reply:
Or even better, just get rid of `name` in `Worker` ? This would then allow `name` from `kwargs`, if I'm not missing anything.

Reply:
Argh, no that won't work either, `name` is part of the spec. I'm really not sure what's the correct way of handling this.

Reply:
Ok, so I took another look and the `name` argument passed to `Worker` is defined and coming from https://github.com/dask/distributed/blob/44b2358e33a0738c4c70ca96db4242636245e07d/distributed/deploy/spec.py#L318-L325 , as an integer. We need to cast it to a string, either here or in dask-cuda. I have no strong feelings but it seems like here would be more appropriate, perhaps someone else would have a similar issue should they use a custom class as well.

Reply:
@mrocklin any news on this PR ?